applications/FluidDynamicsApplication/python_scripts/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_symbolic_generator.py

```diff
@@ -0,0 +1,461 @@
+
+    def _ComputeNonLinearOperator(self, A, H, S, Ug):
+        L = ZeroVector(self.geometry.blocksize)
+        for j in range(self.geometry.ndims):
+            # Convective operator product (A x grad(U))
+            A_j = A[j]
+            H_j = H.col(j)
+            L += A_j * H_j
+            # Diffusive flux
+            # Note that the diffusive flux is not added as it will involve 2nd
+            # order derivatives that vanish when introducing the linear FE
+            # discretization
+
+        # Source term addition
+        L -= S * Ug
+        return L
+
+    def _ComputeNonLinearAdjointOperator(self, A, H, Q, S, Ug, V):
+        L_adj = ZeroVector(self.geometry.blocksize)
+        for j in range(self.geometry.ndims):
+            Q_j = Q.col(j)
+            H_j = H.col(j)
+            # Convective operator product
+            A_j_trans = A[j].transpose()
+            L_adj += A_j_trans * Q_j
+            aux_conv = ZeroMatrix(self.geometry.blocksize, self.geometry.blocksize)
+            for m in range(self.geometry.blocksize):
+                for n in range(self.geometry.blocksize):
+                    A_j_trans_mn = A_j_trans[m, n]
+                    for p in range(self.geometry.blocksize):
+                        aux_conv[m, n] += sympy.diff(A_j_trans_mn, Ug[p]) * H_j[p]
+            L_adj += aux_conv * V
+            # Diffusive operator product
+            # Note that the adjoint diffusive flux is not added as it will
+            # involve 2nd order derivatives that vanish when introducing
+            # the linear FE discretization
+        # Source term addition
+        L_adj += S.transpose() * V
+
+        return L_adj
+
+    def _ComputeVariationalFormulation(self, A, acc, G, H, L_adj, Q, S, Ug, V):
+        # Mass (inertial) term - FE scale (only computed in the implicit case)
+        n1 = ZeroMatrix(1, 1) if self.is_explicit else -V.T*acc
+
+        # Convective term - FE scale
+        conv_flux = ZeroVector(self.geometry.blocksize)
+        for j in range(self.geometry.ndims):
+            conv_flux += A[j] * H.col(j)
+        n2 = - V.transpose() * conv_flux
+
+        # Diffusive term - FE scale
+        n3 = ZeroMatrix(1, 1)
+        for j in range(self.geometry.ndims):
+            for k in range(self.geometry.blocksize):
+                n3[0, 0] += Q[k, j] * G[k, j]
+
+        # Source term - FE scale
+        n4 = V.transpose() * (S * Ug)
+
+        # VMS_adjoint - Subscales
+        subscales = DefineVector('subscales', self.geometry.blocksize)
+        n5 = L_adj.transpose() * subscales
+
+        # Variational formulation (Galerkin functional)
+        self._print(1, " - Compute variational formulation")
+        rv = n1 + n2 + n3 + n4 + n5  # Implicit case with inertial term n1
+
+        return (rv, subscales)
+
+    def _ComputeProjectionsAtGaussPoint(self, acc, bdf, dUdt, f, forcing_terms, H, i_gauss, mg, projections, res, rg, U, Ug, Un, Unn):
```

Comment:
```suggestion
    def _ComputeOSSProjectionsAtGaussPoint(self, acc, bdf, dUdt, f, forcing_terms, H, i_gauss, mg, projections, res, rg, U, Ug, Un, Unn):
```
To make clear which projections we're calculating.